engine/utils/Logging.h

```diff
@@ -41,6 +41,24 @@
 #define DEFINE_VARIANT_TYPE(type, transform) \
     VariantContainer(type) : data(transform) {}
 
+// Assert macros
+#define REPORT_ASSERT_FAILURE(expr, file, line, message)                          \
+    CC_LOG_ERROR("ASSERTION FAILURE: {} IN FILE: {} ON LINE: {} WIH MESSAGE: {}", \
+                 #expr,                                                           \
+                 file,                                                            \
+                 line,                                                            \
+                 message)
+
+// Custom assert macro
+#define CC_ASSERT(expr, message)                                   \
+    if (expr)                                                      \
+    {}                                                             \
+    else                                                           \
+    {                                                              \
+        REPORT_ASSERT_FAILURE(#expr, __FILE__, __LINE__, message); \
+        EXIT_APP                                                   \
+    }
```

Comment:
```suggestion
// Assert macros
#define REPORT_ASSERT_FAILURE(expr, file, line, message)                                        \
     std::cout << "ASSERTION FAILURE: " << #expr << " in file: " << file << " on line: " << line \
               << "\n                        Message: " << message << std::endl

 #define CC_ASSERT(expr, msg)                                                \
     expr || (REPORT_ASSERT_FAILURE(expr, __FILE__, __LINE__, msg) && []() { \
         EXIT_APP;                                                           \
         return true;                                                        \
     }())
```

Reply:
The point is that we want the assertion reporting to be integrated with logging so that assertion errors can be picked out easily. I like what you did with the actual assertion logic though.

Reply:
This code snippet doesn't work. 
```
Expected ')' to match this '('
```

Reply:
Found a workaround. Just needed to wrap `REPORT_ASSERT_FAILURE` in a lambda that returns true.